src/ConstrainedPhotometryModel.cc

```diff
@@ -0,0 +1,213 @@
+                                             double instFlux) const {
+    auto mapping = this->findMapping(ccdImage, "transform");
+    return mapping->transform(measuredStar, instFlux);
+}
+
+void ConstrainedPhotometryModel::getMappingIndices(CcdImage const &ccdImage,
+                                                   std::vector<unsigned> &indices) const {
+    auto mapping = this->findMapping(ccdImage, "getMappingIndices");
+    mapping->getMappingIndices(indices);
+    // TODO: I think I need a for loop here, from the above value to that +mapping->getNpar()?
+}
+
+void ConstrainedPhotometryModel::computeParameterDerivatives(MeasuredStar const &measuredStar,
+                                                             CcdImage const &ccdImage,
+                                                             Eigen::VectorXd &derivatives) const {
+    auto mapping = this->findMapping(ccdImage, "computeParameterDerivatives");
+    mapping->computeParameterDerivatives(measuredStar, measuredStar.getInstFlux(), derivatives);
+}
+
+namespace {
+// Convert photoTransfo's way of storing Chebyshev coefficients into the format wanted by ChebyMap.
+ndarray::Array<double, 2, 2> toChebyMapCoeffs(std::shared_ptr<PhotometryTransfoChebyshev> transfo) {
+    auto coeffs = transfo->getCoefficients();
+    // 4 x nPar: ChebyMap wants rows that look like (a_ij, 1, i, j) for out += a_ij*T_i(x)*T_j(y)
+    ndarray::Array<double, 2, 2> chebyCoeffs = allocate(ndarray::makeVector(transfo->getNpar(), 4));
+    Eigen::VectorXd::Index k = 0;
+    auto degree = transfo->getDegree();
+    for (ndarray::Size j = 0; j <= degree; ++j) {
+        auto const iMax = degree - j;  // to save re-computing `i+j <= degree` every inner step.
+        for (ndarray::Size i = 0; i <= iMax; ++i, ++k) {
+            chebyCoeffs[k][0] = coeffs[j][i];
+            chebyCoeffs[k][1] = 1;
+            chebyCoeffs[k][2] = i;
+            chebyCoeffs[k][3] = j;
+            // std::cout << "idx: " << i << " " << j << " " << k << " " << coeffs[j][i] << " " << std::endl;
+            std::cout << chebyCoeffs[k] << std::endl;
+        }
+    }
+    return chebyCoeffs;
+}
+}  // namespace
+
+std::shared_ptr<afw::image::PhotoCalib> ConstrainedPhotometryModel::toPhotoCalib(
+        CcdImage const &ccdImage) const {
+    std::cout << ccdImage.getName() << std::endl;
+    auto oldPhotoCalib = ccdImage.getPhotoCalib();
+    auto detector = ccdImage.getDetector();
+    auto bbox = detector->getBBox();
+    ChipVisitPhotometryMapping *mapping =
+            dynamic_cast<ChipVisitPhotometryMapping *>(this->findMapping(ccdImage, "toPhotoCalib"));
+    // There should be no way in which we can get to this point and not have a ChipVisitMapping,
+    // so blow up if we don't.
+    assert(mapping != nullptr);
+    auto pixToFocal = detector->getTransform(detector->makeCameraSys(afw::cameraGeom::PIXELS),
+                                             afw::cameraGeom::FOCAL_PLANE);
```

Comment:
I suggest omitting `detector->makeCameraSys` and providing `afw::cameraGeom::PIXELS` directly to `getTransform`. `Detector` methods that take `CameraSys` all have overloaded that handle `CameraSysPrefix` directly (and that assume that same detector, naturally).